numpy/f2py/src/fortranobject.c

```diff
@@ -701,6 +706,15 @@ PyArrayObject* array_from_pyobj(const int type_num,
     }
 
     descr = PyArray_DescrFromType(type_num);
+    /* compatibility with NPY_CHAR */
+    if (type_num == NPY_STRING) {
+        PyArray_DESCR_REPLACE(descr);
+        if (descr == NULL) {
+            return NULL;
+        }
+        descr->elsize = 1;
+        descr->type = NPY_CHARLTR;
```

Comment:
Do we need to keep using `NPY_CHARLTR`? The only difference from `S1` I can see is that that `dtype('c').char` is 'c'.